src/components/Header.tsx

```diff
@@ -15,22 +17,24 @@ export default class Header extends React.Component<HeaderProps, {}> {
   };
 
   render(): JSX.Element {
-    let src = this.props.imgSrc ? require(`${this.props.imgSrc}`) : "";
+    const { imgSrc, alt, text, nav, loggedIn, logOut } = this.props;
+    let src = imgSrc ? require(`${imgSrc}`) : "";
     return(
       <header className="reusable-header clearfix">
-        <span>{this.props.text}</span>
         {
-          this.props.imgSrc &&
-          <img className="header-brand img-rounded" alt={this.props.alt || ""} src={src} />
+          imgSrc &&
+          <img className="header-brand img-rounded" alt={alt || ""} src={src} />
         }
-        { this.props.loggedIn &&
+        <span>{text}</span>
+        { loggedIn &&
           <a
             className="btn header-btn big inverted"
-            href={this.props.logOut}
+            href={logOut}
           >
             <span>Log Out <LogoutIcon /></span>
           </a>
         }
+        { nav && <nav role="navigation">{nav}</nav>}
```

Comment:
Isn't it more common to have the nav menu go before the log out button (i.e. have the log out button be the last thing on the right)?

Reply:
Hmmm it was easier to deal with the css if it was last, but now that I think about it, for tabbing purposes this doesn't work. Will update.